### PR TITLE
UI changes for prayers, unit UI, new gods, warrior priest of sigmar changes, ulric priest

### PR DESCRIPTION
**Pull request by Z3rca**
## Prayer Duration display:
The Map bar now offers a little icon for blessings, showing the name effect and the duration in days (with decimal dot)

## Extended Info display:
Units information is now displayed on the Unit card in the Party menu and the in the troop tree screen. Displaying the maximum health points, amplifications, and additional damage. the additional damage is shown in Proportion based on the wielded weapons. It also shows Text hints for Attributes that have been defined in the tor strings

## Battleprayer book: 
in the Career screen, priest careers, like the warrior priest of sigmar or ulric or damsel can access. The book does provide more information about the battle prayers and allow to select and unselect every prayer likewise the spell book.
The book doesn't appear when the career doesn't have any prayers. 

## Warriorpriest of Ulric as new career
added a selection similar to bloodline selection for the ulric priest. He has 4 battle prayers and allows to apply frost damage. 

## Prestige System
Empire characters can use prestige now to upgrade the weapon shop of the Engineer, as well as can invest in prestigous projects in the empire.

Upgrades: The shop doesn't provide all good weapons directly after finishing the engineer quest, but require 3 updates to gain the full arsenal:
1. upgrade: blunderbuss and long rifles
2. Repeating guns
3. Grenades for blunderbuss & Great cannons

Prestige Projects: In Altdorf, a new character appears, providing at Renown 2 projects the player can invest in to gain more prestige. these are Constructions projects (costing a lot gold) or influence projects. Once all projects are bought the noble allows to exchange influence to Prestige as the player likes.

The projects are thought to be a one time investment, giving a big boost of prestige to a good exchange rate. They should not act as a permanent mild inconvenience exchanging a bit gold to prestige. The projects are extremely expensive as well, yet have a very well exchange to Prestige.

The prestige noble also allows to sell a demi gryph mount for 1000 Prestige. This is done intentional  via a choice dialog not trade, since it should be a one time thing.

  
Obviously the Prestige system can be extended further, but for now I think its in a good state and can be considered done for this update.

## information:

The current info icon and battle prayer button as well as the warrior priest of ulric ability icon and screen are still WIP and are changed later.

### CODE REVIEW

File: CSharpSourceCode/AbilitySystem/CrossHairs/SingleTargetCrosshair.cs
Comment (Z3rca) [inline CSharpSourceCode/AbilitySystem/CrossHairs/SingleTargetCrosshair.cs:1]: one prayer requires that a single target is chosen and spread the effect around it. I saw a lot use in this for future abilities. Spells using single target were not affected

File: CSharpSourceCode/AbilitySystem/Spells/SpellBook/AbilityItemVM.cs
Comment (Z3rca) [inline CSharpSourceCode/AbilitySystem/Spells/SpellBook/AbilityItemVM.cs:1]: abstraction layer reduces code doubling for prayer book

File: CSharpSourceCode/Extensions/AgentExtensions.cs
Comment (Z3rca) [inline CSharpSourceCode/Extensions/AgentExtensions.cs:1]: I think its cleaner to move this logic out of the extensions

File: CSharpSourceCode/HarmonyPatches/DamagePatch.cs
Comment (Z3rca) [inline CSharpSourceCode/HarmonyPatches/DamagePatch.cs:1]: unnecessary object instantiation, i expect maybe a little performance boost

File: CSharpSourceCode/AbilitySystem/Scripts/AxeOfUlricScript.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        protected override void OnInit()
+        {
+            base.OnInit();
+
+            if (!Hero.MainHero.HasCareerChoice("FlameOfUlricKeystone")) return;
+            var comp = Agent.Main.GetComponent<AbilityComponent>();
+            if (comp == null) return;
+
+            var abilities = comp.KnownAbilitySystem.Where(x => x.StringID != Ability.StringID && x.GetCoolDownLeft() > 0);
+            var chosen = abilities.TakeRandom(0).FirstOrDefault();
```
Comment (hunharibo) [inline CSharpSourceCode/AbilitySystem/Scripts/AxeOfUlricScript.cs]: isn't this supposed to be TakeRandom(1)?

File: CSharpSourceCode/AbilitySystem/Spells/Prayers/BattlePrayersVM.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            StatItems.Add(new StatItemVM("Prayer level: ", ((PrayerLevel)highest).ToString()));
+            var prayers = battlePrayers.ConvertAll(input => input.PrayerID);
+            _loreObjectVm = new PrayerLoreObjectVM(this, prayers, Hero.MainHero);
+            
+            foreach (var prayer in _loreObjectVm.PrayerList)
+                if (!prayer.IsKnown)
+                    prayer.IsDisabled = true;
+        }
+
+        private string GetGodCareerIsDevotedTo(CareerObject careerObject)
```
Comment (hunharibo) [inline CSharpSourceCode/AbilitySystem/Spells/Prayers/BattlePrayersVM.cs]: isn't this whole method overriding the actual religion affinity? you can't be sure that they are not worshipping chaos or a different deity even if they have this career.

File: CSharpSourceCode/CampaignMechanics/TORMapInfoVMExtension.cs [hunk 3]
```diff
@@ -19,20 +21,25 @@ public class TORMapInfoVMExtension : BaseViewModelExtension
...
 		private BasicTooltipViewModel _windsHint;
+		
         private BasicTooltipViewModel _cultureResourceHint;
         private BasicTooltipViewModel _artilleryHint;
 		private float _windRechargeRate = 0f;
 		private int _maxWinds = 0;
 		private int _maxArtillery = 0;
 		private int _currentArtilleryItems = 0;
 		private bool _hasCultureResource;
+		private string _remainingBlessingTime;
```
Comment (hunharibo) [inline CSharpSourceCode/CampaignMechanics/TORMapInfoVMExtension.cs]: is this an actual formatting problem, or browser based github is acting up for me?

File: CSharpSourceCode/CharacterDevelopment/CareerSystem/CareerHelper.cs [hunk 4]
```diff
@@ -274,25 +277,61 @@ public static float CalculateTroopWageCareerPerkEffect(TroopRosterElement troop,
             return value;
         }
 
-        public static  bool PlayerOwnsMagicCareer()
+        public static bool IsMagicCareer(CareerObject career)
         {
-            if (Hero.MainHero.HasAnyCareer())
+            if (career == null) return false;
+            
+            if (career == TORCareers.Mercenary||
```
Comment (hunharibo) [inline CSharpSourceCode/CharacterDevelopment/CareerSystem/CareerHelper.cs:284]: Mercenary is magic career?

File: CSharpSourceCode/CampaignMechanics/TORMapInfoVMExtension.cs (hunk 3 above)
Comment (Z3rca) [reply CSharpSourceCode/CampaignMechanics/TORMapInfoVMExtension.cs]: image

i removed spacing, but it looks fine for me

File: CSharpSourceCode/AbilitySystem/Scripts/AxeOfUlricScript.cs (hunk 1 above)
Comment (Z3rca) [reply CSharpSourceCode/AbilitySystem/Scripts/AxeOfUlricScript.cs]: yes fixed

File: CSharpSourceCode/AbilitySystem/Spells/Prayers/BattlePrayersVM.cs (hunk 2 above)
Comment (Z3rca) [reply CSharpSourceCode/AbilitySystem/Spells/Prayers/BattlePrayersVM.cs]: The Menu for the battle prayers  is only be available for actual warrior priest careers. and those are always devoted to a single god Grail Damsel ( of the Lady), Warrior priest (of sigmar ), Warrior priest of Ulric. It doesn't override any god, it just shows to which god the career is devoted to ( and what battle prayers are offered). The Information is a bit redundant, yet the Menu felt a bit empty without it. if this is too confusing I can also remove it.

File: CSharpSourceCode/CharacterDevelopment/CareerSystem/CareerHelper.cs (hunk 4 above)
Comment (Z3rca) [reply CSharpSourceCode/CharacterDevelopment/CareerSystem/CareerHelper.cs:284]: i renamed to "is magic capable career" this is more clear I think. while the Mercenary is the not a magic career perse, we did not remove the ability for them to learn spells.

**Pull request merged**